### ISSUE TITLE
Add "DontParent" option to snapper

### ISSUE THREAD

**Issue by 5H4D0W-X** [Yellow-Dog-Man/Resonite-Issues#1812]
### Is your feature request related to a problem? Please describe.

in some situations (e.g. puzzle pieces or scrabble tiles) a snapper should only move the object it's placed on, without parenting it to the SnapTarget it snapped to.

### Describe the solution you'd like

A boolean input called something like "DontParent" that makes the snapped object move in "world space" instead of reparenting itself and zeroing out its transforms.

### Describe alternatives you've considered

Protoflux that fires on grabbable release and unparents the object after waiting the animation time. When many small pieces are used, this can lead to large numbers of slots.

### Requesters

ShadowX

**Comment by 5H4D0W-X** [comment]
And before anyone points that this will allow objects to be stacked since the SnapTargets never have children, I think that's fine for most use cases and shouldn't be used as an argument against this proposed feature

**Comment by JackTheFoxOtter** [comment]
This would mean that by default a snapped object wouldn't move when the snap target moves. But that can be intentional, I can imagine situations in which this could be useful. It would mean that when configured to not parent, it would instead follow the parenting behavior of the grabbable component. Since the `GrabbableReceiverSurface` already has that option, I think it would make sense to also add that to `SnapTarget`. For consistency I'd suggest flipping the meaning and naming it "ParentPlaced" or "ParentSnapped" to match the option on the `GrabbableReceiverSurface`.

**Comment by Frooxius** [comment]
I don't see how would this work with the current snapping system, because they fundamentally work by parenting the items under the snapping point. This change would go against that design.

We should probably look for alternate approaches to solve your underlying problem here. It sounds like you're needing to snap objects to a grid layout?

**Comment by JackTheFoxOtter** [comment]
Yeah that makes sense. Perhaps a sort of "Grabbable Receiver Point" could work as a generic solution, similar to the existing `GrabbableReceiverSurface`?

**Comment by 5H4D0W-X** [comment]
@frooxius I specifically want to avoid fixed grids. the idea is to have multiple identical objects be able to locate against each other but not necessarily against a grid. 

The puzzle piece analogy explains this pretty well I think. A puzzle piece can be floating anywhere in space and another can always attach to a free side. the important part here is that you can then pick up the first piece without the second following, because it's not parented under the first. this is not currently possible because snappers always parent the snapped item.

What I want to happen is that the snapper visually does the exact same thing as before but without parenting the snapped object to the snapper (i.e converts the snap target's transforms into its parent space and tweens the snapped object based on that).

Granted, this may not lead to the exact same result if the snapper is scaled non-uniformly, but I think that would only apply in niche cases and custom solutions can be found if necessary.

**Comment by 5H4D0W-X** [comment]
> Yeah that makes sense. Perhaps a sort of "Grabbable Receiver Point" could work as a generic solution, similar to the existing `GrabbableReceiverSurface`?

This sounds like a good idea. The existing filtering method on the GrabbableReceiverSurface could be applied here as well.
